### PR TITLE
#163601729 get all offices endpoint 

### PR DESCRIPTION
**Pull request by fantastic-genius**
#### What does this PR do?
Implements get all offices endpoint
#### Description of Task to be completed?
 - implements function to get  office in officeController
- implements route for getting all office endpoint
- implements test for the get offices endpoint in officeApiTest
-refactor how id is generated when party is created in partyController
#### How should this be manually tested?
BASE_URL/api/v1/offices
#### Any background context you want to provide?
None
#### Screenshots (if appropriate)
None

### CODE REVIEW

File: server/controller/partyController.js [hunk 1]
```diff
@@ -69,10 +77,50 @@ class PartyController{
 
         return res.status(400).send({
```
Comment (veeqtor) [inline server/controller/partyController.js]: Try and read up on HTTP statues, so you can know how and when to use the appropriate status code.

File: server/controller/partyController.js [hunk 2]
```diff
@@ -69,10 +77,50 @@ class PartyController{
...
             status: 400,
-            message: "Party not found"
+            error: "Party not found"
         })
         
     }
+
+    editAParty(req, res){
```
Comment (veeqtor) [inline server/controller/partyController.js:85]: Nice work, You will be making a PATCH request to this endpoint, therefore the values to update the party with will be gotten from the JSON body and not the URL params, Please make the necessary changes for this implementation.

File: server/controller/partyController.js [hunk 3]
```diff
@@ -69,10 +77,50 @@ class PartyController{
...
+
+        return res.status(404).send({
+            status: "404",
+            error: "Party doesn't exist"
+        })
+
+        
+    }
+
+    deleteParty(req, res){
```
Comment (veeqtor) [inline server/controller/partyController.js:112]: What happens when I try to delete a non-existing party?

File: server/controller/partyController.js (hunk 2 above)
Comment (fantastic-genius) [reply server/controller/partyController.js:85]: ok,i would refactor my code

File: server/controller/partyController.js (hunk 3 above)
Comment (fantastic-genius) [reply server/controller/partyController.js:112]: Ok i will take that into consideration. Thanks

File: server/controller/partyController.js (hunk 1 above)
Comment (fantastic-genius) [reply server/controller/partyController.js]: Ok. Noted thanks

File: server/controller/partyController.js [hunk 4]
```diff
@@ -75,20 +75,21 @@ class PartyController{
...
+            status: 404,
             error: "Party not found"
         })
         
     }
 
     editAParty(req, res){
-        const id = parseInt(req.params.id);
-        const name = req.params.name;
+        const param_id = parseInt(req.params.id);
```
Comment (veeqtor) [inline server/controller/partyController.js:86]: You are still getting the value from the URL params which is not supposed to be so., Please get your details from the JSON body.

Comment (fantastic-genius) [reply server/controller/partyController.js:86]: Ok

File: server/routes/partyRoutes.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const partyVersionedAPi = "/api/v1/parties/"
 router.post(partyVersionedAPi, partyMiddleware.createPartyMiddleware, partyController.createParty)
 router.get(partyVersionedAPi, partyController.getAllParty)
 router.get(`${partyVersionedAPi}:id`, partyController.getAParty)
-router.patch(`${partyVersionedAPi}:id/:name`, partyMiddleware.editPartyMiddleware, partyController.editAParty)
+router.patch(`${partyVersionedAPi}:id`, partyMiddleware.editPartyMiddleware, partyController.editAParty)
```
Comment (veeqtor) [inline server/routes/partyRoutes.js:11]: I believe this route should be in this format: `/parties/<party-id>/name`

**Pull request merged**